Modules/Loadable/Markups/MRML/CMakeLists.txt

```diff
@@ -29,6 +29,8 @@ set(${KIT}_TARGET_LIBRARIES
   ${MRML_LIBRARIES}
   )
 
+set (${KIT}_INCLUDE_DIRS ${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_CURRENT_BINARY_DIR} CACHE INTERNAL "" FORCE)
```

Comment:
Why is this needed? These include dirs should be added to cache automatically (you can see in Slicer-build\CMakeCache.txt that there are many ...ModuleLogic_INCLUDE_DIRS variables and they are not set using such code).

Reply:
Because the markup node header could not be included in the subject hierarchy plugin cxx without this.

Reply:
It is odd that this is needed, because even without this, my Slicer-build/CMakeCache.txt contains `vtkSlicerMarkupsModuleMRML_INCLUDE_DIRS:INTERNAL=C:/D/S4/Modules/Loadable/Markups/MRML;C:/D/S4R/Slicer-build/Modules/Loadable/Markups/MRML`.

It does not hurt to have this, so if you want you can go on and integrate as is, but it would be nice if you could try a clean build without this line and see if it fails.

Reply:
I started a clean build...

Reply:
It builds. I guess then probably there are several lines of the same kind in Slicer. In any case I removed this one.
@lassoan do you have any ideas what I can do to make it build without a clean build in general? Like re-run CMake or clean build just a few projects etc.

Reply:
I don't think anything can replicate the behavior of a clean build.

I only saw this kind of kit include dir caching line in Subject Hierarchy plugin and maybe one more place. They seemed to be special enough that I did not check if they were still needed, but you are right that those may be removed, too.

Reply:
This is probably an oversight. I suggest to remove it and if there are issues later one, I will have a look.

Reply:
For development it not acceptable having to do a complete 3-5 hour clean build when making such changes in CMake. There must be some kind of way to have a successful build without either adding this line or doing a full clean build...

Reply:
There is really no other way if you want to be 100% sure that you test a clean build without actually doing a clean build. If you are sure that your changes only affect Slicer then you may be able to delete just the Slicer binary folder and then VTK, CTK, etc., may not need to be rebuilt.

I have additional test builds set up for things like this, so I can continue my work, while another build is running in the background.